### PR TITLE
MetalK8s UI: Replace the node metrics by the new chart component

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: UI, montoring

**Context**: 
image

**Summary**:
- embed with the core-ui v0.21.2

**Acceptance criteria**: 

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: ui/src/containers/NodePageMetricsTab.js [hunk 1]
```diff
@@ -81,36 +74,34 @@ const MetricsToggleWrapper = styled.div`
...
   const dispatch = useDispatch();
-  const theme = useTheme();
   const history = useHistory();
   const query = useURLQuery();
   const intl = useIntl();
   const api = useTypedSelector((state) => state.config.api);
-  const metricsTimeSpan = useTypedSelector(
-    (state) => state.app.monitoring.nodeStats.metricsTimeSpan,
-  );
+  const { metricsTimeSpan, value } = useMetricsTimeSpan();
```
Comment (JBWatenbergScality) [inline ui/src/containers/NodePageMetricsTab.js]:
```suggestion
  const { metricsTimeSpan, label } = useMetricsTimeSpan();
```

File: ui/src/services/graphUtils.js [hunk 2]
```diff
@@ -1,79 +1,215 @@
...
   date: Date,
   type: string,
   y: string,
 };
 export type FormattedChartNodesPromRange = Array<FormattedChartNodePromRange>;
+
+// Format the prometheus data to Series
 export const formatNodesPromRangeForChart = (
-  result: Promise<PrometheusQueryResult>[],
+  results: Promise<PrometheusQueryResult>[],
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  results: PrometheusQueryResult[],
```

File: ui/src/services/graphUtils.js [hunk 3]
```diff
@@ -1,79 +1,215 @@
...
+        },
+      };
+    });
+
+    return tempRead.concat(tempWrite);
+  }
+};
+
+export const getSingleResourceSerie = (
+  result: Promise<PrometheusQueryResult>,
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  result: PrometheusQueryResult,
```

File: ui/src/services/graphUtils.js [hunk 4]
```diff
@@ -1,79 +1,215 @@
...
+  nodeIPAddress?: { internalIP: string, name: string },
+  resultAvg?: Promise<PrometheusQueryResult>,
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  resultAvg?:PrometheusQueryResult,
```

File: ui/src/services/graphUtils.js [hunk 5]
```diff
@@ -1,79 +1,215 @@
...
-      temp = matrixResult.result[0].values.map((item) => ({
-        date: fromUnixTimestampToDate(item[0]),
-        type: node.name,
-        y: item[1],
-      }));
-    }
-    return acc.concat(temp);
-  }, []);
-  return reduced;
+): Series => {
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
): Serie[] => {
```

File: ui/src/services/graphUtils.js [hunk 6]
```diff
@@ -1,79 +1,215 @@
...
+  }
+};
+
+export const getSingleResourceSerie = (
+  result: Promise<PrometheusQueryResult>,
+  nodeIPAddress?: { internalIP: string, name: string },
+  resultAvg?: Promise<PrometheusQueryResult>,
+): Series => {
+  const matrixResult: RangeMatrixResult = result?.data?.result[0];
+  const avgMatrixResult = resultAvg ? resultAvg.data.result[0] : [];
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  const avgMatrixResult = resultAvg?.data?.result[0];
```

File: ui/src/services/graphUtils.js [hunk 7]
```diff
@@ -1,79 +1,215 @@
...
+  const prometheusData = matrixResult?.values;
+  const prometheusAvgData = avgMatrixResult?.values;
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  const prometheusData = matrixResult?.values ?? [];
  const prometheusAvgData = avgMatrixResult?.values ?? [];
```

File: ui/src/services/graphUtils.js [hunk 8]
```diff
@@ -1,79 +1,215 @@
...
+  const series = [
+    {
+      data: prometheusData,
+      resource: nodeIPAddress.name,
+      getTooltipLabel: (_, resource: nodeIPAddress.name) => {
+        return nodeIPAddress.name;
+      },
+      getLegendLabel: (_, resource: nodeIPAddress.name) => {
+        return nodeIPAddress.name;
+      },
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
      getTooltipLabel: (_, resource) => {
        return resource;
      },
      getLegendLabel: (_, resource) => {
        return resource;
      },
```

File: ui/src/services/graphUtils.js [hunk 9]
```diff
@@ -1,79 +1,215 @@
...
+  if (resultAvg) {
+    series.push({
+      data: prometheusAvgData,
+      resource: 'Cluster Avg.',
+      getTooltipLabel: (_, resource: 'Cluster Avg.') => {
+        return resource;
+      },
+      getLegendLabel: (_, resource: 'Cluster Avg.') => {
+        return resource;
+      },
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
      getTooltipLabel: (_, resource) => {
        return resource;
      },
      getLegendLabel: (_, resource) => {
        return resource;
      },
```

File: ui/src/services/graphUtils.js [hunk 10]
```diff
@@ -1,79 +1,215 @@
...
+  }
+  return series;
+};
 
-      return acc.concat(tempRead).concat(tempWrite);
-    }, []);
-    return reduced;
+// get the series for symmetrical chart in MetalK8s UI Node page
+export const getSeriesForSymmetricalChart = (
+  results: Promise<PrometheusQueryResult>[],
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  resultsAbove:PrometheusQueryResult,
  resultsBelow:PrometheusQueryResult,

```

File: ui/src/services/graphUtils.js [hunk 11]
```diff
@@ -1,79 +1,215 @@
...
+  nodeIPAddress: { internalIP: string, name: string },
+  metricPrefixes: [],
+  resultsAvg?: Promise<PrometheusQueryResult>[],
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
  resultsAvg?: PrometheusQueryResult[],
```

File: ui/src/services/graphUtils.js [hunk 12]
```diff
@@ -1,79 +1,215 @@
...
+): Series => {
+  const writeRes = results[0];
+  const readRes = results[1];
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]:
```suggestion
```

File: ui/src/services/graphUtils.js [hunk 13]
```diff
@@ -1,79 +1,215 @@
...
+};
 
-      return acc.concat(tempRead).concat(tempWrite);
-    }, []);
-    return reduced;
+// get the series for symmetrical chart in MetalK8s UI Node page
+export const getSeriesForSymmetricalChart = (
+  results: Promise<PrometheusQueryResult>[],
+  nodeIPAddress: { internalIP: string, name: string },
+  metricPrefixes: [],
```
Comment (JBWatenbergScality) [inline ui/src/services/graphUtils.js]: metricsPrefixesAbove: string
metricsPrefixesBelow: string

File: ui/src/components/NodeCPBandwidthChart.js [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+  controlPlaneInterface,
+  showAvg,
+  instanceIP,
+  nodesIPsInfo,
+}: {
+  nodeName: string,
+  controlPlaneInterface: string,
+  showAvg: boolean,
+  instanceIP: string,
+  nodesIPsInfo: Object,
```
Comment (JBWatenbergScality) [inline ui/src/components/NodeCPBandwidthChart.js]: it should be typed

Comment (JBWatenbergScality) [reply ui/src/components/NodeCPBandwidthChart.js]: {[nodeName: string]: {contorlPlane: {ip: string, interface: string}, workloadPlane: {ip: string, interface: string}}}

File: ui/src/components/NodeCPBandwidthChart.js [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+  );
+
+  const nodeCPBandwithOutAvgPrometheusQuery = encodeURIComponent(
+    `${nodesCPBandwidthOutPrometheusQuery.join('+')} / ${
+      nodesCPBandwidthOutPrometheusQuery.length
+    }`,
+  );
+
+  // in
+  const cpBandWidthInPrometheusQuery = `sum(irate(node_network_receive_bytes_total{instance=~"${instanceIP}:${PORT_NODE_EXPORTER}",device="${controlPlaneInterface}"}[5m]))`;
```
Comment (JBWatenbergScality) [inline ui/src/components/NodeCPBandwidthChart.js]: can be factorized with the previous ones

File: ui/src/components/NodeCPBandwidthChart.js [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+  useEffect(() => {
+    if (!isCPBandWidthDataLoading && !showAvg) {
+      // disable avg
+      chartStartTimeRef.current = startTimeRef.current;
+      seriesRef.current = getSeriesForSymmetricalChart(
+        cpBandwidthQuery.data || [],
+        nodeIPAddress,
+        metricPrefixes,
+      );
+    } else if (!isCPBandWidthAvgDataLoading && showAvg) {
```
Comment (JBWatenbergScality) [inline ui/src/components/NodeCPBandwidthChart.js]:
```suggestion
    } else if (!isCPBandWidthDataLoading && !isCPBandWidthAvgDataLoading && showAvg) {
```

File: ui/src/components/NodeCPBandwidthChart.js [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+        cpBandWidthAvgQuery.data,
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCPBandWidthDataLoading, isCPBandWidthAvgDataLoading, showAvg]);
+
+  return (
+    <LineTemporalChart
+      series={seriesRef.current || []}
+      height={160}
```
Comment (JBWatenbergScality) [inline ui/src/components/NodeCPBandwidthChart.js]: It shall probably be a constant shared with other charts

File: ui/src/services/platformlibrary/k8s.js [hunk 18]
```diff
@@ -52,3 +54,184 @@ export const getVolumesCountQuery = (
     enabled: token ? true : false,
   };
 };
+
+// the queries for the metrics
+// TODO: we may want to merge useStartingTimeStamp() and useMetricsTimeSpan(), so the props related to timespan will be returned in one
+export const getCPUUsageQuery = (
+  instanceIP: string,
+  timespanProps: {
```
Comment (JBWatenbergScality) [inline ui/src/services/platformlibrary/k8s.js]: You can extract this timespanProps type so that it can be reused by all the queries

Comment (ChengYanJin) [reply ui/src/services/platformlibrary/k8s.js]: sure. And I am thinking about prehaps I shouldn't put those queries in k8s.js
I will create another folder called `metrics.js`

File: ui/src/components/NodeCPBandwidthChart.js (hunk 14 above)
Comment (ChengYanJin) [reply ui/src/components/NodeCPBandwidthChart.js]: I have added this typing to `NodesState.IPsInfo`

**Review by JBWatenbergScality — DISMISSED** [review]

File: ui/src/containers/NodePageMetricsTab.js [hunk 19]
```diff
@@ -164,20 +170,22 @@ const NodePageMetricsTab = ({
...
-            onChange={(e: SyntheticEvent<HTMLInputElement>) => {
-              writeShowAvg(e.currentTarget.checked);
-              dispatch(
-                updateNodeStatsFetchArgumentAction({
-                  showAvg: e.currentTarget.checked,
-                }),
-              );
-            }}
-          />
+          {instanceIP !== '' && (
```
Comment (JBWatenbergScality) [inline ui/src/containers/NodePageMetricsTab.js]: !instanceIP

Comment (ChengYanJin) [reply ui/src/containers/NodePageMetricsTab.js]: should be `instanceIP` actually ;)

**Comment by ChengYanJin** [comment]
merged in https://github.com/scality/metalk8s/pull/3529

**Pull request closed without merging**